src/main/java/uk/gov/companieshouse/officerfiling/api/service/FilingDataServiceImpl.java

```diff
@@ -70,22 +70,25 @@ public FilingApi generateOfficerFiling(String transactionId, String filingId, St
             if (presentOfficerFilingData.getResignedOn() != null) {
                 //has a removal date so must be a TM01
                 filing.setKind("officer-filing#termination");
+                setTerminationFilingApiData(filing, transactionId, filingId, ericPassThroughHeader);
             } else if (presentOfficerFilingData.getReferenceEtag() == null) {
                 //has no Etag (and has no removal date) so it must be an AP01
                 filing.setKind("officer-filing#appointment");
+                setAppointmentFilingApiData(filing, transactionId, filingId, ericPassThroughHeader);
             } else {
                 throw new NotImplementedException("Kind cannot be calculated using given data for transaction " + transactionId );
             }
         } else {
             filing.setKind("officer-filing#termination");
+            setTerminationFilingApiData(filing, transactionId, filingId, ericPassThroughHeader);
         }
 
-        setFilingApiData(filing, transactionId, filingId, ericPassThroughHeader);
+
         return filing;
     }
 
-    private void setFilingApiData(FilingApi filing, String transactionId, String filingId,
-                                  String ericPassThroughHeader) {
+    private void setTerminationFilingApiData(FilingApi filing, String transactionId, String filingId,
```

Comment:
Do you need the FilingApi filing method param here? Doesn't seem to be used.

Reply:
I am assuming you mean on the setAppointmnetFilingApiData rather than here on Termination,  yes it's needed for Termination, but I have removd it for Appointment.